From f9b3dca644a9220b67d366bdfa56d344b15060d7 Mon Sep 17 00:00:00 2001
From: Kishon Vijay Abraham I <kishon@ti.com>
Date: Fri, 5 Dec 2014 20:26:21 +0530
Subject: [PATCH 1520/1587] mmc: omap_hsmmc: Set DDR bit for UHS_DDR50

commit 5438ad (mmc: omap: clarify DDR timing mode between SD-UHS and eMMC)
replaced UHS_DDR50 with MMC_DDR52 because of which DDR bit is not set in con
register even for cards supporting DDR50. This results in I/O errors.

Fixed it by changing it back to UHS_DDR50.

Fixes: 5438ad95a57c ("mmc: omap: clarify DDR timing mode between SD-UHS and eMMC")

Signed-off-by: Kishon Vijay Abraham I <kishon@ti.com>
Signed-off-by: Sekhar Nori <nsekhar@ti.com>
[zou: Original patch taken from
ti-sdk-am335x-evm-08.00.00.00-Linux-x86-Install.bin]
Signed-off-by: Cao Zou <cao.zou@windriver.com>
---
 drivers/mmc/host/omap_hsmmc.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/mmc/host/omap_hsmmc.c b/drivers/mmc/host/omap_hsmmc.c
index 1753c9d..ee20954 100644
--- a/drivers/mmc/host/omap_hsmmc.c
+++ b/drivers/mmc/host/omap_hsmmc.c
@@ -602,7 +602,7 @@ static void omap_hsmmc_set_bus_width(struct omap_hsmmc_host *host)
 	u32 con;
 
 	con = OMAP_HSMMC_READ(host->base, CON);
-	if (ios->timing == MMC_TIMING_MMC_DDR52)
+	if (ios->timing == MMC_TIMING_UHS_DDR50)
 		con |= DDR;	/* configure in DDR mode */
 	else
 		con &= ~DDR;
-- 
1.7.5.4

